src/gateway.h

```diff
@@ -30,9 +30,12 @@ struct gateway
 	struct raft *raft;           /* Raft instance */
 	struct leader *leader;       /* Leader connection to the database */
 	struct handle *req;          /* Asynchronous request being handled */
+	/* FIXME store this in the req */
 	sqlite3_stmt *stmt;          /* Statement being processed */
+	/* FIXME store this in the req */
 	bool stmt_finalize;          /* Whether to finalize the statement */
```

Comment:
Note that both `stmt` and `stmt_finalize` are cross-request values, basically because we need batch rows yielded by a query (otherwise the might be to many and the client has no way to cancel). What happens in those cases is that the client sends a first request containing the query statement (either an existing prepared statement or plain SQL), then it gets back immediately a response with the first batch of resulting rows, and then the client has to submit additional requests for each additional batch for rows, which will be delivered as individual response. Hope that's clear.

Reply:
Thanks for pointing that out, I'll remove the FIXMEs.